pkgs/applications/system/tango-controls/default.nix

```
@@ -0,0 +1,101 @@
+{ cmake
+, cppzmq
+, doxygen
+, fetchurl
+, graphviz
+, lib
+, libjpeg_turbo
+, libsodium
+, mariadb-connector-c
+, openjdk11
+, pkg-config
+, python3
+, stdenv
+, systemd
+, zeromq
+, zlib
+}:
+let
+  # omniorb is in nixpkgs, but tango doesn't support the latest version
+  omniorb_4_2 = stdenv.mkDerivation rec {
+    pname = "omniorb";
+    version = "4.2.5";
+
+    src = fetchurl {
+      url = "mirror://sourceforge/project/omniorb/omniORB/omniORB-${version}/omniORB-${version}.tar.bz2";
+      sha256 = "1fvkw3pn9i2312n4k3d4s7892m91jynl8g1v2z0j8k1gzfczjp7h";
+    };
+
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = [ python3 ];
+
+    enableParallelBuilding = true;
+    hardeningDisable = [ "format" ];
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "tango-controls";
+  version = "9.4.2";
+
+  src = fetchurl {
+    url = "https://gitlab.com/api/v4/projects/24125890/packages/generic/TangoSourceDistribution/${version}/tango-${version}.tar.gz";
+    hash = "sha256-3KQYBNd450gBRqJBa0SHXWXIADfpdNPeBLskXGTPqBY=";
+  };
+
+  enableParallelBuilding = true;
+  nativeBuildInputs = [ cmake pkg-config ];
+  buildInputs = [
+    zlib
+    omniorb_4_2
+    zeromq
+    cppzmq
+    libjpeg_turbo
+    mariadb-connector-c
+    libsodium
+    doxygen
+    # needed for the docs
+    graphviz
+  ] ++ lib.optional (!stdenv.isDarwin) systemd;
+  propagatedBuildInputs = [ omniorb_4_2 cppzmq zeromq libjpeg_turbo ];
+
+  cmakeFlags = [
+    "-DMySQL_LIBRARY_RELEASE=${mariadb-connector-c}/lib/mariadb/libmariadb${stdenv.hostPlatform.extensions.sharedLibrary}"
+    "-DMySQL_INCLUDE_DIR=${mariadb-connector-c.dev}/include/mariadb"
+    "-DMySQL_EXECUTABLE=${mariadb-connector-c}/bin/mariadb"
+    "-DCMAKE_VERBOSE_MAKEFILE=TRUE"
+    "-DCMAKE_SKIP_RPATH=ON"
+    "-DTSD_TAC=ON"
+    "-DTSD_JAVA_PATH=${openjdk11}"
+  ];
+
+  # To build proper systemd services for Tango, we need a way to tell when the DB service is started.
+  # This patch adds systemd support for this.
+  patches = lib.optional (!stdenv.isDarwin) ./sd_notify_cmake.patch;
```

```suggestion
  patches = lib.optional stdenv.isLinux ./sd_notify_cmake.patch;
```
For example, BSD doesn't support systemd.

---

Good point, fixed!